### DISCUSSION TITLE
add swim trasformer module

### DISCUSSION THREAD

**Discussion by CesareDavidePace** [obss/sahi#789] [Q&A] [answered]
Hi, i added a swim trasformer block to yolov5 architecture and i have this error

File ~/.local/lib/python3.8/site-packages/sahi/auto_model.py:63, in AutoDetectionModel.from_pretrained(model_type, model_path, model, config_path, device, mask_threshold, confidence_threshold, category_mapping, category_remapping, load_at_init, image_size, **kwargs)
     60 model_class_name = MODEL_TYPE_TO_MODEL_CLASS_NAME[model_type]
     61 DetectionModel = import_model_class(model_type, model_class_name)
---> 63 return DetectionModel(
     64     model_path=model_path,
     65     model=model,
     66     config_path=config_path,
     67     device=device,
     68     mask_threshold=mask_threshold,
     69     confidence_threshold=confidence_threshold,
     70     category_mapping=category_mapping,
     71     category_remapping=category_remapping,
     72     load_at_init=load_at_init,
...
     38 def autopad(k, p=None, d=1):  # kernel, padding, dilation
     39     # Pad to 'same' shape outputs
     40     if d > 1:

ModuleNotFoundError: No module named 'models'

I added the model to both common.py and yolo.py of yolov5. If I do inference with the normal function yolo, but with sahi I get this.
What else do I need to change?

**Comment by fcakyon** [comment, marked as answer]
SAHI does not support custom yolov5 architectures. You can implement DetectionModel class for your use case.

**Reply by CesareDavidePace** [reply]
perfect, I already fixed it. Thanks!

**Comment by nakuda1** [comment]
Hi, sorry to bother you, but can you share how you figured it out? When I use my own yolov5, use my own .pt document, it does not work. Thanks.

**Reply by CesareDavidePace** [reply]
i can make a PR, maybe
